patches/api/0496-WIP-DataComponent-API.patch

```diff
@@ -0,0 +1,4152 @@
++    @NullMarked
++    @ApiStatus.Experimental
++    @ApiStatus.NonExtendable
++    interface ApplyStatusEffects extends ConsumeEffect {
++
++        /**
++         * Effect instances to grant
++         *
++         * @return effect
++         */
++        List<PotionEffect> effects();
++
++        /**
++         * Float between 0 and 1, chance for the effect to be applied.
++         *
++         * @return chance
++         */
++        float probability();
++    }
++}
+diff --git a/src/main/java/io/papermc/paper/datacomponent/item/consumable/ItemUseAnimation.java b/src/main/java/io/papermc/paper/datacomponent/item/consumable/ItemUseAnimation.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..8cd6dbe4ea5ee3270b9428a9c29cbd88823d9f6c
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/datacomponent/item/consumable/ItemUseAnimation.java
+@@ -0,0 +1,17 @@
++package io.papermc.paper.datacomponent.item.consumable;
++
++/**
++ * Represents the hand animation that is used when a player is consuming this item.
++ */
++public enum ItemUseAnimation {
++    NONE,
++    EAT,
++    DRINK,
++    BLOCK,
++    BOW,
++    SPEAR,
++    CROSSBOW,
++    SPYGLASS,
++    TOOT_HORN,
++    BRUSH
++}
+diff --git a/src/main/java/io/papermc/paper/item/MapPostProcessing.java b/src/main/java/io/papermc/paper/item/MapPostProcessing.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..5843768d0be2ae4a0219636ed7640727808da567
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/item/MapPostProcessing.java
+@@ -0,0 +1,6 @@
++package io.papermc.paper.item;
++
++public enum MapPostProcessing {
++    LOCK,
++    SCALE
++}
+diff --git a/src/main/java/io/papermc/paper/registry/RegistryKey.java b/src/main/java/io/papermc/paper/registry/RegistryKey.java
+index 9b39e33514b15a9d07104e2ad826d0da11f569d6..116857b4479565b602b94d227ee32dc29ebd6e5f 100644
+--- a/src/main/java/io/papermc/paper/registry/RegistryKey.java
++++ b/src/main/java/io/papermc/paper/registry/RegistryKey.java
+@@ -1,5 +1,6 @@
+ package io.papermc.paper.registry;
+ 
++import io.papermc.paper.datacomponent.DataComponentType;
+ import net.kyori.adventure.key.Keyed;
+ import org.bukkit.Art;
+ import org.bukkit.Fluid;
+@@ -124,6 +125,11 @@ public sealed interface RegistryKey<T> extends Keyed permits RegistryKeyImpl {
+      * @see io.papermc.paper.registry.keys.SoundEventKeys
+      */
+     RegistryKey<Sound> SOUND_EVENT = create("sound_event");
++    /**
++     * Built-in registry for data component types.
++     * @see io.papermc.paper.registry.keys.DataComponentTypeKeys
++     */
++    RegistryKey<DataComponentType> DATA_COMPONENT_TYPE = create("data_component_type");
+ 
+ 
+ 
+diff --git a/src/main/java/io/papermc/paper/util/Filtered.java b/src/main/java/io/papermc/paper/util/Filtered.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..54e1a383c4eeb595417ea6e9e4e9527a8ffd9a99
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/util/Filtered.java
```

Comment:
Util doesn't feel like the right package for this. It's not a general utility, it's a Minecraft type